src/compile/projection/parse.ts

```diff
@@ -0,0 +1,65 @@
+import {SHAPE, X, Y} from '../../channel';
+import {Config} from '../../config';
+import {MAIN} from '../../data';
+import {isProjectionFieldDef} from '../../fielddef';
+import {GEOSHAPE} from '../../mark';
+import {Projection} from '../../projection';
+import {contains} from '../../util';
+import {VgProjection} from '../../vega.schema';
+import {Model} from '../model';
+import {UnitModel} from '../unit';
+import {ProjectionComponent} from './component';
+
+export function parseProjection(model: Model, config: Config) {
+  if (model instanceof UnitModel) {
+    model.component.projection = parseUnitProjection(model, config);
+  } else {
+    const mergedProjection = parseNonUnitProjections(model);
+  }
+}
+
+function parseUnitProjection(model: UnitModel, config: Config): ProjectionComponent {
```

Comment:
No need to take `config` as parameter.  You can access it from `model.config`. 
The init part already passed `config` to all models in the model tree.